rust/connlib/tunnel/src/lib.rs

```diff
@@ -70,16 +70,20 @@ where
     CB: Callbacks + 'static,
 {
     pub fn poll_next_event(&mut self, cx: &mut Context<'_>) -> Poll<Result<Event<GatewayId>>> {
-        let Some(device) = self.device.as_mut() else {
+        if self.device.is_none() {
             self.no_device_waker.register(cx.waker());
             return Poll::Pending;
         };
 
         match self.role_state.poll_next_event(cx) {
             Poll::Ready(Event::SendPacket(packet)) => {
-                device.write(packet)?;
+                self.device.as_mut().unwrap().write(packet)?;
```

Comment:
Why change this? Isn't `let-else` much cleaner than unwrapping everywhere?

Reply:
The problem is, that since we need to call `self.update_interface()` we're borrowing `self` twice as mutable using `let-else`

Reply:
Can we refactor `update_interface` then to pass in whatever we are borrowing mutably? Or make it a free-function? I don't think the current design is a good trade-off.